File: 2024-2025/geometri.py
import math

def main():
    while True:
        print("\033c", end="")  # This clears the console
        print("Choose a geometric shape to calculate the area:")
        print("1. Triangle")
        print("2. Rectangle")
        print("3. Circle")
        print("4. Exit")
        choice = input()

        if choice == "1":
            calculate_triangle_area()
        elif choice == "2":
            calculate_rectangle_area()
        elif choice == "3":
            calculate_circle_area()
        elif choice == "4":
            return
        else:
            print("Invalid choice. Please enter a number between 1 and 4.")

        print("Do you want to calculate another area? (y/n)")
        if input().lower() != "y":
            break

def calculate_triangle_area():
    base_triangle = float(input("Enter the base of the triangle: "))
    height = float(input("Enter the height of the triangle: "))

    area = triangle_area(base_triangle, height)
    print(f"The area of the triangle is: {area}")

def calculate_rectangle_area():
    length = float(input("Enter the length of the rectangle: "))
    width = float(input("Enter the width of the rectangle: "))

    area = rectangle_area(length, width)
    print(f"The area of the rectangle is: {area}")

def calculate_circle_area():
    radius = float(input("Enter the radius of the circle: "))

    area = circle_area(radius)
    print(f"The area of the circle is: {area}")

def triangle_area(base_triangle, height):
    return 0.5 * base_triangle * height

def rectangle_area(length, width):
    return length * width

def circle_area(radius):
    return math.pi * radius ** 2

if __name__ == "__main__":
    main()

File: 2024/geometri.cs
using System;

namespace Geometri
{
    class Program
    {
        static void Main(string[] args)
        {
            while (true)
            {
                Console.Clear();
                Console.WriteLine("Choose a geometric shape to calculate the area:");
                Console.WriteLine("1. Triangle");
                Console.WriteLine("2. Rectangle");
                Console.WriteLine("3. Circle");
                Console.WriteLine("4. Exit");
                string choice = Console.ReadLine();

                switch (choice)
                {
                    case "1":
                        CalculateTriangleArea();
                        break;
                    case "2":
                        CalculateRectangleArea();
                        break;
                    case "3":
                        CalculateCircleArea();
                        break;
                    case "4":
                        return;
                    default:
                        Console.WriteLine("Invalid choice. Please enter a number between 1 and 4.");
                        break;
                }

                Console.WriteLine("Do you want to calculate another area? (y/n)");
                if (Console.ReadLine().ToLower() != "y")
                {
                    break;
                }
            }
        }

        static void CalculateTriangleArea()
        {
            Console.WriteLine("Enter the base of the triangle:");
            double baseTriangle = Convert.ToDouble(Console.ReadLine());
            Console.WriteLine("Enter the height of the triangle:");
            double height = Convert.ToDouble(Console.ReadLine());

            double area = TriangleArea(baseTriangle, height);
            Console.WriteLine($"The area of the triangle is: {area}");
        }

        static void CalculateRectangleArea()
        {
            Console.WriteLine("Enter the length of the rectangle:");
            double length = Convert.ToDouble(Console.ReadLine());
            Console.WriteLine("Enter the width of the rectangle:");
            double width = Convert.ToDouble(Console.ReadLine());

            double area = RectangleArea(length, width);
            Console.WriteLine($"The area of the rectangle is: {area}");
        }

        static void CalculateCircleArea()
        {
            Console.WriteLine("Enter the radius of the circle:");
            double radius = Convert.ToDouble(Console.ReadLine());

            double area = CircleArea(radius);
            Console.WriteLine($"The area of the circle is: {area}");
        }

        static double TriangleArea(double baseTriangle, double height)
        {
            return 0.5 * baseTriangle * height;
        }

        static double RectangleArea(double length, double width)
        {
            return length * width;
        }

        static double CircleArea(double radius)
        {
            return Math.PI * Math.Pow(radius, 2);
        }
    }
}
